### ISSUE TITLE
demo process-list open files a lot!

### ISSUE THREAD

**Issue by cuiyudaniuHT** [libvmi/libvmi#484]
Demo process-list will open a lot of sockets to qemu when insert to a "while" function. This is because that  process-list will call "kvm_test" function. And in kvm_test (libvmi/driver/kvm/kvm.c),  "kvm_destroy" is only called at last. But most time, "return VMI_SUCCESS" will directly return which means the new allocated vmi is not released.

We added "kvm_destroy(vmi);" before all "return" to FIX this bug.

<img width="1236" alt="_20170601121828" src="https://cloud.githubusercontent.com/assets/29113339/26664808/0487b6d0-46c7-11e7-9170-8289c492b9ea.png">

![image](https://cloud.githubusercontent.com/assets/29113339/26664828/458fabd8-46c7-11e7-8436-aca26a405af1.png)

**Comment by tklengyel** [comment]
Indeed, there should be a `kvm_destroy` before each return after `kvm_init` suceeds. Would you mind opening PR to fix it?
